pyranges/get_fasta.py

```diff
@@ -107,3 +116,116 @@ def get_fasta(gr, path=None, pyfaidx_fasta=None):
+
+    Parameters
+    ----------
+    gr : PyRanges
+
+        Coordinates.
+
+    group_by : str or list of str
+    
+        intervals are grouped by this/these ID column(s): these are exons belonging to same transcript
+
+    path : str
+
+        Path to fasta file. It will be indexed using pyfaidx if an index is not found
+
+    pyfaidx_fasta : pyfaidx.Fasta
+
+        Alternative method to provide fasta target, as a pyfaidx.Fasta object
+
+
+    Returns
+    -------
+    DataFrame
+
+        Pandas DataFrame with a column for Sequence, plus ID column(s) provided with "group_by"
+
+    Note
+    ----
+
+    This function requires the library pyfaidx, it can be installed with 
+    ``conda install -c bioconda pyfaidx`` or ``pip install pyfaidx``.
+
+    Sorting the PyRanges is likely to improve the speed.
+    Intervals on the negative strand will be reverse complemented.       
+
+    Warning
+    -------
+
+    Note that the names in the fasta header and gr must be the same.
+
+    See also
+    --------
+    get_sequence : obtain sequence of single intervals
+
+
+    Examples
+    --------
+
+    >>> gr = pr.from_dict({"Chromosome": ['chr1', 'chr1', 'chr1'],
+    ...                    "Start": [0, 9, 18], "End": [4, 13, 21],
+                           "Strand":['+', '-', '-'],
+                           "transcript": ['t1', 't2', 't2']})
+    >>> gr
+    +--------------+-----------+-----------+--------------+--------------+
+    | Chromosome   |     Start |       End | Strand       | transcript   |
+    | (category)   |   (int32) |   (int32) | (category)   | (object)     |
+    |--------------+-----------+-----------+--------------+--------------|
+    | chr1         |         0 |         4 | +            | t1           |
+    | chr1         |         9 |        13 | -            | t2           |
+    | chr1         |        18 |        21 | -            | t2           |
+    +--------------+-----------+-----------+--------------+--------------+
+    Stranded PyRanges object has 3 rows and 5 columns from 1 chromosomes.
+    For printing, the PyRanges was sorted on Chromosome and Strand.
+
+
+    >>> tmp_handle = open("temp.fasta", "w+")
+    >>> _ = tmp_handle.write(">chr1\\n")
+    >>> _ = tmp_handle.write("AAACCCTTTGGGAAACCCTTTGGG\\n")
+    >>> tmp_handle.close()
+  
+    >>> seq = pr.get_transcript_sequence(gr, path="temp.fasta", group_by='transcript')
+    >>> seq
+
+      transcript Sequence
+    0         t1     AAAC
+    1         t2  AAATCCC
+
+    To write to a file in fasta format:
+    >>> with open('outfile.fasta', 'w') as fw:
+    ...     nchars=60
+    ...     for row in seq.itertuples():
+    ...         s = '\\n'.join([ row.Sequence[i:i+nchars] for i in range(0, len(row.Sequence), nchars)])
+    ...         fw.write(f'>{row.transcript}\\n{s}\\n')
+
+    """
+
+
+    if gr.stranded:
+        gr = gr.sort('5')        
+        #is_minus = z.Strand=='-'
```

Comment:
We should remove comments before submitting. (I know the code base isn't perfect in this regard)